client/my-sites/email/form/mailboxes/index.ts

```diff
@@ -57,7 +57,26 @@ class MailboxForm< T extends EmailProvider > {
 		}
 	}
 
+	hasErrors() {
+		return Object.values( this.formFields ).some( ( field ) => field.error !== null );
+	}
+
+	hasValidValues() {
+		return Object.values( this.formFields as MailboxFormFields )
+			.filter( ( field ) => field.required ?? false )
```

Comment:
It seems we're always setting `required` when creating `MailboxFormField` objects so we could simplify this code:

```suggestion
			.filter( ( field ) => field.required )
```

Reply:
> It seems we're always setting `required `when creating `MailboxFormField` ..

Not always, as the boolean fields don't set it. Another issue is that the typed state of the object cannot be enforced by Typescript at run time, hence the object may be modified by the state and other JS code parts.

Reply:
>  the typed state of the object cannot be enforced by Typescript at run time

Maybe something like the following would help then?

```suggestion
			.filter( ( field as MailboxFormFieldBase ) => field.required )
```